addons/sale_payment/models/payment.py

```diff
@@ -63,7 +63,9 @@ def _confirm_so(self):
             _logger.info('<%s> transaction authorized, auto-confirming order %s (ID %s)', self.acquirer_id.provider, self.sale_order_id.name, self.sale_order_id.id)
             if self.sale_order_id.state in ('draft', 'sent'):
                 self.sale_order_id.with_context(send_email=True).action_confirm()
-        elif self.state == 'done':
+            return 'authorized_sale'
```

Comment:
Returned codes are prefixed. If you look at them you have pay_sale_ ... or pay_invoice_ ... I would stick on that convention if possible.

Anyway looking at the code you change the return of the method. Currently it returns something if there is an error, nothing if there isn't any. In your case you make the method return something although it is a success. I do not like that kind of API change.

Isn't it possible to move the authorize management in the caller ?